packages/block-library/src/post-excerpt/edit.js

```diff
@@ -32,13 +33,40 @@ export default function PostExcerptEditor( {
 } ) {
 	const isDescendentOfQueryLoop = Number.isFinite( queryId );
 	const userCanEdit = useCanEditEntity( 'postType', postType, postId );
-	const isEditable = userCanEdit && ! isDescendentOfQueryLoop;
-
 	const [
 		rawExcerpt,
 		setExcerpt,
 		{ rendered: renderedExcerpt, protected: isProtected } = {},
 	] = useEntityProp( 'postType', postType, 'excerpt', postId );
+
+	// Check if the post type supports excerpts.
+	let postTypeSupportsExcerpts = useSelect( ( select ) =>
+		postType
+			? !! select( coreStore ).getPostType( postType )?.supports.excerpt
+			: false
+	);
```

Comment:
A few notes:

* The `useSelect` needs a `postType` dependency.
* We can return `true` early when `postType === page` from the selector.
* There's no need for ternary. Casting the return value to `boolean` will do the trick.

The end results will be something like this:

```js
// Check if the post type supports excerpts.
const postTypeSupportsExcerpts = useSelect(
	( select ) => {
		if ( postType === 'page' ) {
			return true;
		}

		return !! select( coreStore ).getPostType( postType )?.supports
			.excerpt;
	},
	[ postType ]
);
```